test/php/library/Icingadb/ProvidedHook/Reporting/SlaTest.php

```diff
@@ -0,0 +1,338 @@
+                ['event_time' => 1500000, 'hard_state' => 0, 'previous_hard_state' => 2]
+            ],
+            'downtime' => [
+                ['downtime_id' => $this->makeId(), 'downtime_start' => 1100000, 'downtime_end' => 1300000],
+                ['downtime_id' => $this->makeId(), 'downtime_start' => 1400000, 'downtime_end' => 1600000]
+            ],
+        ]);
+
+        // 1000..1100: OK, no downtime
+        // 1100..1200: OK, in downtime
+        // 1200..1300: CRITICAL, in downtime
+        // 1300..1400: CRITICAL, no downtime (only period counting for SLA, -10%)
+        // 1400..1500: CRITICAL, in downtime
+        // 1500..1600: OK, in downtime
+        // 1600..2000: OK, no downtime
+        $timeline = $this->report->getSlaTimeline($this->start, $this->end, 'host');
+        $this->assertSame(90.0, $timeline->getResult());
+
+        $timeline = $this->report->getSlaTimeline($this->start, $this->end, 'service');
+        $this->assertSame(90.0, $timeline->getResult());
+    }
+
+    public function testCriticalBeforeInterval()
+    {
+        $this->insertSlaHistoryEvents(
+            ['state' => [['event_time' => 0, 'hard_state' => 2, 'previous_hard_state' => 99]]]
+        );
+
+        // If there is no event within the SLA interval, the last state from before the interval should be used.
+        $timeline = $this->report->getSlaTimeline($this->start, $this->end, 'host');
+        $this->assertSame(0.0, $timeline->getResult());
+
+        $timeline = $this->report->getSlaTimeline($this->start, $this->end, 'service');
+        $this->assertSame(0.0, $timeline->getResult());
+    }
+
+    public function testCriticalBeforeIntervalWithDowntime()
+    {
+        $this->insertSlaHistoryEvents([
+            'state'    => [['event_time' => 800000, 'hard_state' => 2, 'previous_hard_state' => 99]],
+            'downtime' => [['downtime_id' => $this->makeId(), 'downtime_start' => 600000, 'downtime_end' => 1500000]]
+        ]);
+
+        $timeline = $this->report->getSlaTimeline($this->start, $this->end, 'host');
+        $this->assertSame(50.0, $timeline->getResult());
+
+        $timeline = $this->report->getSlaTimeline($this->start, $this->end, 'service');
+        $this->assertSame(50.0, $timeline->getResult());
+    }
+
+    public function testCriticalBeforeIntervalWithOverlappingDowntimes()
+    {
+        $this->insertSlaHistoryEvents([
+            'state'    => [['event_time' => 800000, 'hard_state' => 2, 'previous_hard_state' => 99]],
+            'downtime' => [
+                ['downtime_id' => $this->makeId(), 'downtime_start' => 600000, 'downtime_end' => 1000000],
+                ['downtime_id' => $this->makeId(), 'downtime_start' => 800000, 'downtime_end' => 1200000],
+                ['downtime_id' => $this->makeId(), 'downtime_start' => 1000000, 'downtime_end' => 1400000],
+                ['downtime_id' => $this->makeId(), 'downtime_start' => 1600000, 'downtime_end' => 2000000], // -20%
```

Comment:
The -20% comment here on a downtime line makes little sense as downtimes never reduce the availability percentage.

Reply:
The comment should indicate the time period between the current downtime start event and the previous downtime end event. `downtime_end => 1400.....downtime_start => 1600`

Reply:
That's not really related to that specific downtime but rather with a gap between the downtimes. That should be clearer when keeping the comment similar to how it was before: https://github.com/Icinga/icingadb/blob/4ed4db3ab642622bb917797a8207589ce5f89f07/tests/sql/sla_test.go#L91-L99